include/sys/mm.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_MM_H
+#define ZEPHYR_INCLUDE_SYS_MM_H
+
+#include <sys/util.h>
+
+/*
+ * Caching mode definitions. These are mutually exclusive.
+ */
+
+/** No caching. Most drivers want this. */
+#define K_MAP_CACHE_NONE	0
+
+/** Write-through caching. Used by certain drivers. */
+#define K_MAP_CACHE_WT		1
+
+/** Full write-back caching. Any RAM mapped wants this. */
+#define K_MAP_CACHE_WB		2
+
+/** Reserved bits for cache modes in k_map() flags argument */
+#define K_MAP_CACHE_MASK	(BIT(3) - 1)
+
+/*
+ * Region permission attributes. Default is read-only, no user, no exec
+ */
+
+/** Region will have read/write access (and not read-only) */
+#define K_MAP_RW		BIT(3)
+
+/** Region will be executable (normally forbidden) */
+#define K_MAP_EXEC		BIT(4)
+
+/** Region will be accessible to user mode (normally supervisor-only) */
+#define K_MAP_USER		BIT(5)
+
+#ifndef _ASMLANGUAGE
+#include <stdint.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_MMU
```

Comment:
I think it'd be better *not* to have such #ifdefs in header files for function prototypes, since it prevents using `IS_ENABLED(CONFIG_MMU)` in the calling code. I.e. leave it up to the linker to throw the error of a non-existing function rather than the compiler. An additional benefit is that you don't need to add exceptions to the documentation building to make sure these get included included there.

Reply:
OK, removed.